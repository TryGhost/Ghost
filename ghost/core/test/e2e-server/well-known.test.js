const {agentProvider, matchers, mockManager} = require('../utils/e2e-framework');
const {anyString, anyEtag} = matchers;

describe('.well-known', function () {
    let agentGhostAPI;

    before(async function () {
        agentGhostAPI = await agentProvider.getGhostAPIAgent();
    });

    after(function () {
        mockManager.restore();
    });

    describe('GET /jwks.json', function () {
        it('should return a JWKS', async function () {
            await agentGhostAPI
                .get('/.well-known/jwks.json')
                .expectStatus(200)
                .matchBodySnapshot({
                    keys: [{
                        kid: anyString,
                        n: anyString
                    }]
                })
                .matchHeaderSnapshot({
                    etag: anyEtag
                });
        });
    });
});
