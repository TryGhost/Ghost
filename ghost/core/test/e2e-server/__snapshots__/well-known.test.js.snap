// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.well-known GET /jwks.json should return a JWKS 1: [body] 1`] = `
Object {
  "keys": Array [
    Object {
      "e": "AQAB",
      "kid": Any<String>,
      "kty": "RSA",
      "n": Any<String>,
      "use": "sig",
    },
  ],
}
`;

exports[`.well-known GET /jwks.json should return a JWKS 2: [headers] 1`] = `
Object {
  "cache-control": "public, max-age=86400",
  "content-length": "277",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;
